rules/jsll/tests/jsll-validate-config/test.ts

```diff
@@ -0,0 +1,155 @@
+    },
+    {
+        name: `The init script is an external script and the config is not valid'`,
+        reports: [{ message: messages.missingRequiredConfigProps }, { message: messages.missingUseShortNameForContentBlob }],
+        serverConfig: {
+            '/': generateHTMLPage(`${code.jsllScript}<script src="init.js"></script>`),
+            '/init.js': { content: `console.log('adsf');${code.emptyObjconfig}` }
+        }
+    },
+    {
+        name: `"awa.init" has no arguments`,
+        reports: [{ message: messages.noConfigArgs }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(null, code.noConfigArgs)}`)
+    },
+    {
+        name: `"config" misses required properties "autoCapture"`,
+        reports: [{ message: messages.missingAutoCapture }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(deleteProp('autoCapture'), code.initConfig)}`)
+    },
+    {
+        name: `"appId" is missing`,
+        reports: [{ message: messages.missingAppId }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(deleteProp('appId'), code.initConfig)}`)
+    },
+    {
+        name: `"appId" is not a string`,
+        reports: [{ message: messages.invalidAppId }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(modifyConfigVal('appId', 1), code.initConfig)}`)
+    },
+    {
+        name: `"autoCapture" is not a valid object`,
+        reports: [{ message: messages.invalidAutoCapture }, { message: messages.missingLineageAndScroll }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(modifyConfigVal('autoCapture', true), code.initConfig)}`)
+    },
+    {
+        name: `"config" is an empty object`,
+        reports: [{ message: messages.missingRequiredConfigProps }, { message: messages.missingUseShortNameForContentBlob }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(null, code.emptyObjconfig)}`)
+    },
+    {
+        name: `"config" properties are strings instead of identifiers`,
+        serverConfig: generateHTMLPage(`${scriptWrapper(`var config=${JSON.stringify(stringPropertyConfig)};`, code.initConfig)}`)
+    },
+    {
+        name: `Both required and optional properties string literals`,
+        reports: [{ message: messages.missingAppId }, { message: messages.missingPageNameAndEnv }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(deleteProp(['appId', 'pageName', 'env']), code.initConfig)}`)
+    },
+    {
+        name: `"config" misses optional properties`,
+        reports: [{ message: messages.missingOptionalConfigProp }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(deleteProp('useShortNameForContentBlob'), code.initConfig)}`)
+    },
+    {
+        name: `"market" has an invalid value`,
+        reports: [{ message: messages.invalidMarketValue }],
+        serverConfig: generateHTMLPage(`${scriptWrapper(modifyConfigVal('market', 'market'), code.initConfig)}`)
+    },
+    {
```

Comment:
Shouldn't you add a more complex `market` example to make sure it passes? Similar to what the Xbox page does?

Reply:
@molant  Done. I added a more complex `market` example in `calculatedConfig` in `common.ts`.